### DISCUSSION TITLE
brew doesn't detect latest version of xcode CLT

### DISCUSSION THREAD

**Discussion by lmBored** [Homebrew/discussions#5690] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 35ebf4a047b00b2b3de33ba0081d71293cdadede
Last commit: 34 hours ago
Core tap JSON: 22 Oct 09:20 UTC
Core cask tap JSON: 22 Oct 09:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 16.0
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Your Command Line Tools are too outdated.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 16.0.


Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  neofetch

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libusb-1.0.0.dylib
  /usr/local/lib/libusb-1.0.dylib
```

### Description of issue

When I run `brew upgrade`, I get this error:

```
Error: Your Command Line Tools are too outdated.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 16.0.
```

Firstly, I tried running
```
sudo rm -rf /Library/Developer/CommandLineTools
sudo xcode-select --install
```

but that gives me the error that I don't have the folder `CommandLineTools` in `/Library/Developer`. Digging further, it seems like later version of xcode is installed in `/Applications/Xcode.app` instead:

```
$ xcode-select -p
/Applications/Xcode.app/Contents/Developer
```

Therefore, I downloaded `clt for xcode 16.0` straight from [https://developer.apple.com/download/all/](https://developer.apple.com/download/all/). However, I still get the above warnings as well as error. Could someone help me with this please? 

Addition: also is it true that in later versions, xcode doesn't install themself in `/Library/Developer/CommandLineTools` anymore? I see many posts online still use that path but I don't have it.

**Comment by lmBored** [comment]
Oh forgot to mention, I am using macOS Sequoia 15.0.1

**Comment by ZhongRuoyu** [comment]
> that gives me the error that I don't have the folder `CommandLineTools` in `/Library/Developer`

From this, you may not have installed CLT properly. Are you sure you downloaded "Command Line Tools for Xcode 16" instead of "Xcode 16"? These are not the same thing. CLT installs to `/Library/Developer/CommandLineTools`, whereas Xcode installs to `/Applications/Xcode.app` by default.

**Comment by carlocab** [comment]
I just removed `/Library/Developer/CommandLineTools` and installed `Command Line Tools for Xcode 16` from https://developer.apple.com/download/all/. It was installed into `/Library/Developer/CommandLineTools`.

```
❯ /Library/Developer/CommandLineTools/usr/bin/clang --version
Apple clang version 16.0.0 (clang-1600.0.26.3)
Target: arm64-apple-darwin24.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
❯ pkgutil --pkg-info com.apple.pkg.CLTools_macOS_SDK
package-id: com.apple.pkg.CLTools_macOS_SDK
version: 16.0.0.0.1.1724870825
volume: /
location: /
install-time: 1729592037
❯ date +%s
1729592325
```

**Comment by lmBored** [comment]
This is super weird, I tried installing "Command Line Tools for Xcode 16" again but it still doesn't work. Then I uninstall xcode, reinstall with homebrew, install "Command Line Tools for Xcode 16" manually, and I finally have `/Library/Developer/CommandLineTools` now!

Thank you for all the help!
